core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -70,19 +72,40 @@ class KafkaHealthcheck(private val brokerId: Int,
   }
 
   /**
-   *  When we get a SessionExpired event, we lost all ephemeral nodes and zkclient has reestablished a
-   *  connection for us. We need to re-register this broker in the broker registry.
+   *  When we get a SessionExpired event, it means that we have lost all ephemeral nodes and ZKClient has re-established
+   *  a connection for us. We need to re-register this broker in the broker registry.
    */
-  class SessionExpireListener() extends IZkStateListener {
+  class SessionExpireListener extends IZkStateListener {
+
+    private[server] val stateToSensorMap = {
+      import KeeperState._
+      val stateToNameMap = Map(
+        Disconnected -> "disconnected",
+        SyncConnected -> "sync-connected",
+        AuthFailed -> "auth-failed",
+        ConnectedReadOnly -> "connected-read-only",
+        SaslAuthenticated -> "sasl-authenticated",
+        Expired -> "expired"
+      )
+      stateToNameMap.map { case (state, stateName) =>
+        val sensorName = s"zk-$stateName"
+        val sensor = metrics.sensor(sensorName)
+        sensor.add(metrics.metricName(s"$sensorName-rate", "zk", s"The average number of zk $sensorName events per second"),
+          new Rate(new Count))
+        state -> sensor
+      }
+    }
+
     @throws(classOf[Exception])
-    def handleStateChanged(state: KeeperState) {}
+    def handleStateChanged(state: KeeperState) {
```

Comment:
It probably makes sense to be consistent across the methods though and I see that we are using `override` in one of the other methods. So I'll at least change them to be consistent (one way or the other).